src/client.rs

```diff
@@ -47,21 +49,123 @@ pub struct Client {
     inner: Arc<ClientRef>,
 }
 
-impl Client {
-    /// Constructs a new `Client`.
-    pub fn new() -> ::Result<Client> {
-        let mut client = try_!(new_hyper_client());
-        client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
-        Ok(Client {
+impl From<NativeTlsClient> for Client {
+    fn from(tls_client: NativeTlsClient) -> Client {
+        let mut hyper_client = ::hyper::Client::with_connector(
+            ::hyper::client::Pool::with_connector(
+                Default::default(),
+                ::hyper::net::HttpsConnector::new(tls_client)));
+
+        hyper_client.set_redirect_policy(::hyper::client::RedirectPolicy::FollowNone);
+
+        Client {
             inner: Arc::new(ClientRef {
-                hyper: RwLock::new(client),
+                hyper: RwLock::new(hyper_client),
                 redirect_policy: Mutex::new(RedirectPolicy::default()),
                 auto_referer: AtomicBool::new(true),
                 auto_ungzip: AtomicBool::new(true),
             }),
+        }
+    }
+}
+
+/// A `ClientBuilder` can be used to a `Client` with a custom TLS configuration:
+///
+/// - with hostname verification disabled
+/// - with one or multiple custom certificates
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// extern crate hyper_native_tls;
+/// extern crate reqwest;
+///
+/// use std::fs::File;
+/// use std::io::Read;
+///
+/// use hyper_native_tls::Certificate;
+/// use reqwest::ClientBuilder;
+///
+/// fn main() {
+///     // get a client builder
+///     let mut client_builder = ClientBuilder::new().unwrap();
+///
+///     // read a local binary DER encoded certificate
+///     let mut buf = Vec::new();
+///     let _ = File::open("my-cert.der").unwrap().read_to_end(&mut buf).unwrap();
+///
+///     // create a certificate
+///     let cert = Certificate::from_der(&buf).unwrap();
+///
+///     // add the certificate
+///     client_builder.add_root_certificate(cert).unwrap();
+///
+///     // create the actual client
+///     let client = client_builder.build().unwrap();
+///
+///     // [...]
+/// }
+/// ```
+///
+pub struct ClientBuilder {
+    inner: NativeTlsClientBuilder,
+    hostname_verification: bool,
+}
+
+impl ClientBuilder {
+    /// Constructs a new `ClientBuilder`
+    pub fn new() -> ::Result<ClientBuilder> {
+        let inner_tls_client = try_!(
+            NativeTlsClient::builder()
+                .map_err(|e| ::hyper::Error::Ssl(Box::new(e))));
+        Ok(ClientBuilder {
+            inner: inner_tls_client,
+            hostname_verification: true,
         })
     }
 
+    /// Returns a client
+    pub fn build(self) -> ::Result<Client> {
+        let mut tls_client = try_!(
+            self.inner.build().map_err(|e| ::hyper::Error::Ssl(Box::new(e))));
+        if ! self.hostname_verification {
+            tls_client.dannger_disable_hostname_verification(true);
```

Comment:
[Yes](https://github.com/sfackler/hyper-native-tls/blob/master/src/lib.rs#L142) probably a typo, I'll make a PR to fix this.